### PR TITLE
fixes #768 fix race condition in generation of table map

### LINKED ISSUE #768: Handle TableOps exists() issues
This method has concurrency issues, mainly due to cacheing and the distributed environment of Accumulo.  For example, one process could create a table and this method could return false for another process because it's zookeeper cache doesn't have the table ID yet.  Another example... one process calls exists, gets true then performs an action on that table but the table is deleted before the process can perform the action on that table.  We seem to have to explain this to users quite often.  At the very least, we should document this in javadoc.

### PR DESCRIPTION
**Pull request by keith-turner**

### CODE REVIEW

File: core/src/main/java/org/apache/accumulo/core/client/impl/TableMap.java [hunk 1]
```diff
@@ -98,4 +106,8 @@ public TableMap(Instance instance, ZooCache zooCache) {
   public Map<String,String> getIdtoNameMap() {
     return tableIdToNameMap;
   }
+
+  public boolean isCurrent(ZooCache zc) {
+    return this.zooCache == zc && this.updateCount == zc.getUpdateCount();
+  }
```
Comment (milleruntime) [inline core/src/main/java/org/apache/accumulo/core/client/impl/TableMap.java]: This method is nice.  Would it be better to use equals comparison?

File: core/src/main/java/org/apache/accumulo/core/client/impl/Tables.java [hunk 2]
```diff
@@ -132,22 +111,35 @@ public static String getTableName(Instance instance, String tableId)
...
     TableMap map;
+
+    final ZooCache zc = getZooCache(instance);
+
+    map = getTableMap(instance, zc);
+
+    if (!map.isCurrent(zc)) {
+      instanceToMapCache.invalidate(instance.getInstanceID());
+      map = getTableMap(instance, zc);
+    }
```
Comment (milleruntime) [inline core/src/main/java/org/apache/accumulo/core/client/impl/Tables.java]: Do we need to synchronize on the TableMap cache?  Or does guava handle that?

Comment (keith-turner) [reply core/src/main/java/org/apache/accumulo/core/client/impl/Tables.java]: Guava handles that

File: core/src/main/java/org/apache/accumulo/core/client/impl/TableMap.java (hunk 1 above)
Comment (keith-turner) [reply core/src/main/java/org/apache/accumulo/core/client/impl/TableMap.java]: In this case I wanted to make sure it was the exact same zoocache instance

Comment (milleruntime) [reply core/src/main/java/org/apache/accumulo/core/client/impl/TableMap.java]: Ah right, gotcha.

**Pull request merged**